browser/ipfs/ipfs_tab_helper_unittest.cc

```diff
@@ -716,10 +787,66 @@ TEST_F(IpfsTabHelperUnitTest, GatewayIPNS_NoRedirect_WhenNoDnsLinkRecord) {
       GURL("https://ipfs.io/ipns/brantly.eth/page?query#ref"));
 
   auto headers = net::HttpResponseHeaders::TryToCreate("HTTP/1.1 200 OK");
-  helper->MaybeCheckDNSLinkRecord(headers.get());
+  helper->MaybeCheckDNSLinkRecord(headers.get(), false);
 
   EXPECT_TRUE(ipfs_host_resolver()->resolve_called());
   ASSERT_EQ(GURL(), helper->GetIPFSResolvedURL());
 }
 
+TEST_F(IpfsTabHelperUnitTest, DetectPageLoadingError_ShowInfobar) {
+  const GURL url(
+      "https://ipfs.io/ipns/"
+      "k2k4r8ni09jro03sto91pyi070ww4x63iwub4x3sc13qn5pwkjxhfdt4");
+  const GURL redirected_to_url(
+      "ipns://k2k4r8ni09jro03sto91pyi070ww4x63iwub4x3sc13qn5pwkjxhfdt4");
+
+  SetIpfsCompanionEnabledFlag(false);
+
+  bool is_fallback_showed = false;
+  ipfs_tab_helper()->SetSetShowFallbackInfobarCallbackForTesting(
+      base::BindLambdaForTesting([&](const GURL& current_url) {
+        is_fallback_showed = true;
+        EXPECT_EQ(current_url, url);
+      }));
+
+  web_contents()->SetOnDidFinishNavigationCompleted(
+      base::BindLambdaForTesting([&](content::NavigationHandle* handler) {
+        ipfs_tab_helper()->SetAutoRediretCallbackForTesting(
+            base::BindLambdaForTesting([&](const GURL& url_to_check) {
+              EXPECT_EQ(redirected_to_url, url_to_check);
+              return handler;
+            }));
+        ipfs_tab_helper()->LoadUrlForAutoRedirect(redirected_to_url);
```

Comment:
Looks like this tests are trying to do the same thing as browser tests but in more complicated way